apps/remote_control/lib/lexical/remote_control/code_intelligence/ast.ex

```diff
@@ -0,0 +1,101 @@
+defmodule Lexical.RemoteControl.CodeIntelligence.Ast do
+  alias Lexical.Document
+  alias Lexical.Document.Position
+
+  def original_module(aliases_mapping, module) do
```

Comment:
I'm not sure this function should be here or not.

if you think this is the right place, I can add some tests like:

```elixir
  describe "original_module/2" do
    test "it should return the original module" do
      aliases_mapping = %{
        User => Project.Structs.User,
        :__MODULE__ => MyModule
      }

      assert Ast.original_module(aliases_mapping, User) == Project.Structs.User
    end

    test "it should return the original when the module is a nested" do
      aliases_mapping = %{
        User => Project.Structs.User,
        :__MODULE__ => MyModule
      }

      assert Ast.original_module(aliases_mapping, User.Profile) == Project.Structs.User.Profile
    end

    test "it should return the current module" do
      aliases_mapping = %{
        User => Project.Structs.User,
        :__MODULE__ => MyModule
      }

      assert Ast.original_module(aliases_mapping, :__MODULE__) == MyModule
    end

    test "it should return the module when the module is not in aliases mapping" do
      aliases_mapping = %{
        User => Project.Structs.User,
        :__MODULE__ => MyModule
      }

      assert Ast.original_module(aliases_mapping, Project.XX) == Project.XX
    end
  end

```